c_common/models/data_speed_up_packet_gatherer/src/data_speed_up_packet_gatherer.c

```diff
@@ -324,51 +340,93 @@ static inline uint calculate_sdram_address_from_seq_num(uint seq_num) {
                     * sizeof(uint);
 }
 
+static inline void schedule_timeout(void) {
+    wait_until = time + TIMEOUT;
+    log_debug("scheduled timeout for %u (now %u)", wait_until, time);
+}
+
+static inline void cancel_timeout(void) {
+    wait_until = 0;
+}
+
+static inline void set_message_length(const void *end) {
+    my_msg.length = ((const uint8_t *) end) - &my_msg.flags;
+    if (my_msg.length > 272) {
+        log_error("bad message length %u", my_msg.length);
+    }
+}
+
 //! \brief searches through received seq nums and transmits missing ones back
 //! to host for retransmission
 static void process_missing_seq_nums_and_request_retransmission(void) {
+    //! \brief Used to guard access to the received_seq_nums_store from this
+    //!   function; it counts the number of running calls to this function.
+    //!   Access to this variable is only allowed when you have disabled
+    //!   interrupts!
+    static uint access_lock = 0;
+
+    uint sr;
+    sr = spin1_irq_disable();
+    if (++access_lock == 1) {
```

Comment:
OK, I think this now means you can never get in to the function.  Should this be a post-increment, so you are looking at the value before it was incremented in the comparison?